server/aesdsocket.c

```diff
@@ -35,377 +41,643 @@
+                        {
+                            goto close_filefd;
+                        }                        
+                        lseek(file_fd, 0, SEEK_SET );
+
+                        if(echo_file_socket(file_fd,thread_params->fd)==-1)
+                        {
+                            goto close_filefd;
+                        }
+                        char time_str[80];
+                    
+                        if(dequeue(time_str)==0)
+                        {
+                            if(write_data(file_fd,time_str,strlen(time_str))!=-1)
+                            {
+                                newline_data += strlen(time_str);
+                            }
+                        }
+                        start_ptr = newline_data;
+                        close(file_fd);
+                        status = pthread_mutex_unlock(&s_data.mutex);
+                        if(status != 0)
+                        {
+                            syslog(LOG_ERR,"Mutex Unlock: %s",strerror(errno));
+                            goto unlock_mutex;
+                        }
+                        break;
+                    }
+                }
+                buffer_length += recv_bytes;
+                thread_params->thread_type = recv_t;
+                break;
+            
+            case accept_t:
+            	break;
+            case join_t:
+            	break;
+        }
+    }
+    close_filefd: close(file_fd);
+    unlock_mutex: pthread_mutex_unlock(&s_data.mutex);
+    free_mem: free(buffer);
+    free_socket_fd: close(thread_params->fd);
+                    thread_params->thread_completed = true;
+                    syslog(LOG_DEBUG,"Closed connection from %s",thread_params->ip_addr);
+                    return 0;
+}
 
-	if(status == -1)
-	{
-		syslog(LOG_ERR, "bind error = %d\n",errno);
-		fprintf(stderr, "bind error: %d\n", errno);
-		exit(-1);
-	}
-	else
-	{
-		syslog(LOG_DEBUG, "bind success\n");
-	}
-	//	break;
-	//}
-		
-	//freeaddrinfo(servinfo);
-	
-	//if(p == NULL)
-	//{
-	//	fprintf(stderr, "server: failed to bind\n");
-	//	exit(-1);
-	//}
 
 
-    	if(argc>1)
+int main(int argc,char **argv)
+{
+	init_socket();
+	init_queue();
+	bool d_arg = false;
+	thread_type_t thread_type;
+	if(argc>1)
 	{
 		if(strcmp(argv[1],"-d") == 0)
 		{
 			d_arg = true;
 		}
 	}
+	int status=0,yes=1;
+	struct addrinfo hints;
+	struct addrinfo *p = NULL;
+	char s[INET6_ADDRSTRLEN];
+	memset(s,0,sizeof(s));
+	struct sockaddr_storage c;
+	socklen_t addr_size = sizeof(c);
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_socktype = SOCKET_TYPE;
+	hints.ai_flags = AI_PASSIVE;
+
+    	status = getaddrinfo(NULL, PORT, &hints, &s_data.host_addr_info);
+    	if(status != 0)
+	{
+		syslog(LOG_ERR, "getaddrinfo error = %d\n",errno);
+		fprintf(stderr, "getaddrinfo error: %d\n", errno);
+		return -1;
+	}
+    	s_data.free_address_info = true;
 
-	//if(!fork())
```

Comment:
@abaa3023 rather than comment out large chunks of code you should delete with a commit which you could revert later as needed.  Let revision control take care of history for you